torchgeo/trainers/segmentation.py

```diff
@@ -115,6 +126,11 @@ def __init__(self, **kwargs: Any) -> None:
                 UserWarning,
             )
         self.ignore_index = kwargs["ignore_index"]
+
+        self.class_weights = kwargs.get("class_weights", None)
+        if not isinstance(self.class_weights, (list, type(None))):
+            raise ValueError("class_weights must be a List or None")
```

Comment:
I prob wouldn't bother with manual type checking, that's what mypy is for

Reply:
Is it okay if I take out lines 131 & 132?

Reply:
Yep